.travis.yml

```diff
@@ -1,18 +1,22 @@
-dist: xenial
-language: python
-python:
-  - 3.6
+language: c
+env:
+    matrix:
+        - DOCKER_IMAGE=nmstate/centos7-nmstate-dev TEST_TYPE=integ
+        - DOCKER_IMAGE=nmstate/fedora-nmstate-dev TEST_TYPE=integ
+        - DOCKER_IMAGE=nmstate/fedora-nmstate-dev TEST_TYPE=unit_code_style
+        - DOCKER_IMAGE=nmstate/fedora-nmstate-dev TEST_TYPE=unit_py27
+        - DOCKER_IMAGE=nmstate/fedora-nmstate-dev TEST_TYPE=unit_py36
 
 addons:
   apt:
     packages:
-    - libgirepository1.0-dev
+        - realpath
 
-install:
-  - pip install tox
+services:
+    - docker
 
-script:
-  - tox -e flake8,pylint,check-py27,check-py36
+before_install:
+    - docker pull $DOCKER_IMAGE
 
-after_success:
-  - tox -e coveralls
+script:
+    - sudo ./automation/run-tests.sh
```

Comment:
Why `sudo` is needed?

Reply:
Let me remove it and see how it goes.
It's copy for old travis config I guess travis now enforcing root privilege now.